GGBRW/BOOLR
# Fix wire bugs

This pull request fixes problems connecting wires to wires

How to reproduce:
- connect input 1 to output 1 
- connect input 0 to output 0
- connect wire connecting i/o 1 to i/o 0 (from top to bottom)
- activate inputs 0 and 1
- deactivate input 1

<img width="400" alt="Screenshot 2020-09-06 at 09 34 31" src="https://user-images.githubusercontent.com/12446853/92319756-4d06c680-f024-11ea-861e-4f757a9f9b6d.png">

When deactivating Input#1 I would expect no wires to change value, but upper part of the circuit goes off, while the lower stays on:
<img width="400" alt="Screenshot 2020-09-06 at 09 35 34" src="https://user-images.githubusercontent.com/12446853/92319773-7293d000-f024-11ea-9304-c9f2d9b53636.png">
But disabling Input#2 works as expected

Other fixed problems:
1. If I try to add wire with opposite direction and disable both Input#1 and Input#2, then all wires will still be on  due to loop powering of two wires of opposite direction
2. Removing wires does not trigger update
3. Undoing wires removal does not trigger update
4. Removing selection and undoing does not display wire but adds it
5. When multiple wires connected to one port only the last one is really connected, so when dragging the component/port earlier wires are not moving
6. When multiple wire connected to one port and one of them goes off it will turn off the port no matter what other wires' value is
7. When connecting wire to port that already has wire, than first wire will be connected to the second wire, not the port
